interface/src/app/mlmodel/selector.tpl.html

```diff
@@ -1,8 +1,8 @@
 <div class="alert alert-danger" role="alert" ng-if="$ctrl.errorMessage">
   {{$ctrl.errorMessage}}
 </div>
-<select
-  ng-options="option as option.title for option in $ctrl.availableMlModels"
-  ng-model="$ctrl.selectedMlModel">
-  <option value="">--- Please select a machine learning model ---</option>
+<select name="selectMlModel"
+  ng-options="opt.title for opt in $ctrl.availableMlModels track by opt.id"
```

Comment:
It definitely does not work if I remove `track by opt.id`. I suspect the reason is that the `select` directive may be trying to compare the entire `ng-model` *object* with what it finds in `$ctrl.availableMlModels` and it may be working with a copy of the object somewhere so they don't match. Using `track by opt.id` tells it exactly how to match the selection with the `ng-model` so it works. Just a hunch, though, without digging into the Angular source for the `select` directive.